src/monodroid/jni/embedded-assemblies-zip.cc

```diff
@@ -65,18 +76,8 @@ EmbeddedAssemblies::zip_load_entries (int fd, const char *apk_name, monodroid_sh
 	bool runtime_config_blob_found = false;
 #endif // def NET6
 
-	bool bundled_assemblies_slow_path = false;
-	if (bundled_assemblies.empty ()) {
-		// We're called for the first time, let's allocate space for all the assemblies counted during build. resize()
-		// will allocate all the memory and zero it in one operation
-		bundled_assemblies.resize (application_config.number_of_assemblies_in_apk);
-	} else {
-		// We are probably registering some assemblies dynamically, possibly loading from another apk, we need to extend
-		// the storage instead of taking advantage of the pre-allocated slots.
-		bundled_assemblies_slow_path = true;
-	}
-
-	int64_t assembly_count = application_config.number_of_assemblies_in_apk;
+	bool bundled_assemblies_slow_path = bundled_assembly_index >= application_config.number_of_assemblies_in_apk;
+	uint32_t max_assembly_name_size = application_config.bundled_assembly_name_width - 1;
```

Comment:
This is "differently" amusing: we have to do `+1` to support the terminating NUL, yet when we hit native code we just `-1` that value.

Do we actually *need* the terminating NUL?

Probably, actually?

The `set_entry_data<false>()` method uses `max_assembly_name_size` as the `len` parameter to **strncpy**(3), and `strncpy()` won't read or write past `len`, it *feels like* there could be a situation where `XamarinAndroidBundledAssembly::name` isn't properly NUL-terminated?

Reply:
The NUL is just a failsafe, just as is `max_assembly_name_size`.